### DISCUSSION TITLE
Refactoring Helper - HelperConfig.s.sol & no run function?

### DISCUSSION THREAD

**Discussion by RamanSB** [Cyfrin/foundry-full-course-cu#1555] [Q&A]
In an earlier episode we learnt that all Solidity scripts (<fileName>.**s.sol**) should contain a run function how comes we do not adhere to that in the HelperConfig.s.sol script?

OR is it the case where that is only true for Deployment Scripts?

**Comment by EngrPips** [comment]
Hello @RamanSB , your script might not have a run function if you choose to not have it. But run function makes it very easy to use your script from the command line as whenever you call that script from the command line, the run function would be the entry point into the script and from their chains of call can be made. but if you are using a script from another script like we did with `helperConfig.sol` it is not necessary you have a `run` function
